### ISSUE TITLE
Render <FormErrorMessage> at render time instead of dispatch time

### ISSUE THREAD

**Issue by AndreasJacobsen** [chakra-ui/chakra-ui#3240]
# 🐛 Bug report

The `<FormErrorMessage>`-component only adds the error message to the DOM at message dispatch time instead of at site render time. This causes some screen readers to not see the aria-live attributes present when they appear. I tested this using VoiceOver and NVDA, the aria-live messages worked great in VoiceOver but failed using NVDA. I do not have access to test this on JAWS. 

I have encountered this issue before, some assistive technologies fail to detect aria-live regions that are added to a page dynamically. 

MDN tells us 
`"Including an aria-live attribute or a specialized live region role (such as role="alert") on the element you want to announce changes to works as long as you add the attribute before the changes occur — either in the original markup, or dynamically using JavaScript."`. I'm not sure what they mean by "dynamically" but they state that the element should have the attribute before the change occours. 

I found some more info on this here https://almerosteyn.com/2017/09/aria-live-regions-in-react
## 💥 Steps to reproduce

> Create a form
> Have your errors in <FormErrorMessage/ >
> Form error messages with aria-live are not available in the DOM until they are dispatched. 

## 🧐 Expected behavior

> Form error messages are loaded into the DOM at site render time and visually hidden using CSS and if needed aria-hidden (depending on how this is solved)

**Comment by with-heart** [comment]
Just wanted to call out this part of the article you linked:

> One of the most important things to keep in mind when making use of an ARIA live region is that changes to the content will only be read out by screen readers if the entire region already exists in the DOM and there has been any changes.
>
> This means we cannot render the live region in the DOM for the first time at the moment we want to dispatch a message to the screen reader. No, on the initial render the inside text is typically ignored by screen readers.
>
> And here comes the caveat in a React application; we need to ensure that the live region is already rendered when we send our first message and that it stays rendered until we no longer require it.
>
> In a React application it means that the live region needs to exist as high up in the component tree as possible and then we can use the same region for all communication.

I was unaware of the requirement that the live region must already exist in the DOM, so that's good to know. 

I don't have time to test atm but it also appears that screenreaders don't queue changes made to multiple live regions at the same time and only read the latest update, so we might need something a little more robust in order to provide the best a11y experience here for cases where forms are validating multiple fields at once.

**Comment by AndreasJacobsen** [comment]
As far as I know it is not possible to make screen readers announce multiple aria-live="assertive" regions, only one. I don't know if this was intended but I can see some logic in this. 

Say we have a really long form and 20 wrong entries when we press submit, do we want users to hear 20 messages? And even worse have to press the ctrl-key to abort the 20 messages 20 times? 

For longer forms I usually solve this by creating an error dialog that contains all the errors with role="alert" and putting it at the top of the form, onSubmit I move focus to this alert. But this method is probably way too opinionated to be used in Chakra (besides, how would you gauge what is a "long" form?) 

Now let me preface by saying that I know a bunch about accessibility, but I am not a very good developer (working on that), so take my suggestions as from someone with a good understanding of accessibility but a surface level understanding of React. 

Possible solutions I can think if 
1: Render every error message at page load time, and visually hide using CSS.
Remember that things hidden using visibility: hidden, display: non e or the HTML5 attribute hidden are also hidden from the browser accessibility API. So visually hiding using this method may not solve this (but I have not tested this yet). 
2: render empty tags ( example `<span aria-live="assertive"> </span>`) for the error message that have aria-live tags in them. This will, theoretically, allow screen readers to see that there is a live region here and look for changes. Without having thought much about it I would also append aria-atomic="true" to these regions to ensure that all the elements in the region are announced. 

Regarding global shared regions, that is still a bit over my level. But I see nothing but advantages in loading live regions as early as possible. I dove deeper into the google machine to find more documentation on this, but good sources were very hard to find. Not even Googles developer docs about aria-live mention this, I think this could be a good opportunity for Chakra to stand out and also document this change - so that hopefully others learn from it.

**Comment by k-hoang** [comment]
@with-heart 
Yeh, multiple live regions don't play well with each other. Ideally <FormErrorMessage> should not use `aria-live` and instead  be associated to the element with `aria-describedby` . On top of this, there should be a separate live region component with all the errors to be used in one place for the entire form. This component needs to be permanently mounted so that screen readers are aware of any changes.

**Comment by AndreasJacobsen** [comment]
Please don't close this issue stalebot :)

**Comment by AndreasJacobsen** [comment]
Please don't close this issue stalebot :)

**Comment by AndreasJacobsen** [comment]
> Hi!
> This issue hasn't seen any activity recently. We close inactive issues after 35 days to manage the volume of issues we receive.
> If we missed this issue or you want to keep it open, please reply here. That will reset the timer and allow more time for this issue to be addressed before it is closed.
> 

Still an active issue stalebot

**Comment by hpersson** [comment]
It is very easy with the current behavior to create a form that shifts around after hitting "submit" because the FormErrorText takes up additional space. It would be much easier to have the FormErrorText be rendered at all times and toggle visibility based off of isInvalid. +1 to this suggestion!

In addition to being an accessibility issue, I think this also greatly increases the friction of building consistent UX for forms with error fields.
